BrokerClientTest.shouldThrowExceptionOnTimeout is flaky

**Summary**

- How often does the test fail?
  - Happened only once on the `main` branch. 
- Does it block your work?
  -  Currently no.
- Do we suspect that it is a real failure?
  - No, it seems like a timing issue.

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
Error:  Tests run: 9, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 23.48 s <<< FAILURE! -- in io.camunda.zeebe.gateway.broker.BrokerClientTest
Error:  io.camunda.zeebe.gateway.broker.BrokerClientTest.shouldThrowExceptionOnTimeout -- Time elapsed: 0.774 s <<< FAILURE!
java.lang.AssertionError: 

Expecting a throwable with root cause being an instance of:
  java.util.concurrent.TimeoutException
but was an instance of:
  io.camunda.zeebe.gateway.cmd.PartitionNotFoundException: Expected to execute command on partition 1, but either it does not exist, or the gateway is not yet aware of it
	at io.camunda.zeebe.gateway.impl.broker.BrokerRequestManager.determineBrokerNodeIdProvider(BrokerRequestManager.java:214)
	at io.camunda.zeebe.gateway.impl.broker.BrokerRequestManager.sendRequestInternal(BrokerRequestManager.java:118)
	at io.camunda.zeebe.gateway.impl.broker.BrokerRequestManager.lambda$sendRequestInternal$1(BrokerRequestManager.java:106)
	...(6 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
	at io.camunda.zeebe.gateway.broker.BrokerClientTest.shouldThrowExceptionOnTimeout(BrokerClientTest.java:240)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
</pre>
</details>

**Hypotheses**

/

**Logs**

[LOGS](https://github.com/camunda/zeebe/actions/runs/6690402311/job/18175651783)

--- a/gateway/src/test/java/io/camunda/zeebe/gateway/broker/BrokerClientTest.java
+++ b/gateway/src/test/java/io/camunda/zeebe/gateway/broker/BrokerClientTest.java
@@ -51,6 +51,7 @@ import java.util.List;
 import java.util.concurrent.TimeoutException;
 import org.agrona.DirectBuffer;
 import org.agrona.concurrent.UnsafeBuffer;
+import org.awaitility.Awaitility;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Rule;
@@ -104,6 +105,9 @@ public final class BrokerClientTest {
           topology.addBrokerIfAbsent(0);
           topology.setBrokerAddressIfPresent(0, stubAddress.toString());
         });
+    Awaitility.await("Topology is updated")
+        .untilAsserted(
+            () -> assertThat(topologyManager.getTopology().getPartitions()).isNotEmpty());
 
     client =
         new BrokerClientImpl(

